tests/trainer/test_trainer.py

```diff
@@ -949,7 +948,39 @@ def test_num_sanity_val_steps_neg_one(tmpdir, limit_val_batches):
         max_steps=1,
     )
     assert trainer.num_sanity_val_steps == float('inf')
-    val_dataloaders = model.val_dataloader__multiple()
+
+
+@pytest.mark.parametrize(['limit_val_batches', 'expected'], [
+    pytest.param(0, 0),
+    pytest.param(5, 7),
+])
+def test_num_sanity_val_steps_progress_bar(tmpdir, limit_val_batches, expected):
+    """
+    Test val_progress_bar total with "num_sanity_val_steps" Trainer argument.
+    """
+    class CustomCallback(Callback):
+        def __init__(self):
+            self.val_progress_bar_total = 0
+
+        def on_validation_epoch_end(self, trainer, pl_module):
+            self.val_progress_bar_total += trainer.progress_bar_callback.val_progress_bar.total
```

Comment:
I see a small issue with this test. 
This relies on the fact that the progress bar callback gets registered in the trainer in a "special" way. 
If we refine our callback system in the future, we will have to rewrite this test. I'll think of a better way, if there exists one :)

Reply:
I can do this by overriding the `ProgressBar` instead of `Callback` if it sounds good.

Reply:
ok, sounds good!